src/Blocks/BlockCauldron.h

```diff
@@ -47,11 +67,33 @@ class cBlockCauldronHandler :
 				if (Meta > 0)
 				{
 					a_ChunkInterface.SetBlockMeta(a_BlockX, a_BlockY, a_BlockZ, --Meta);
-					a_Player.GetInventory().RemoveOneEquippedItem();
-					a_Player.GetInventory().AddItem(cItem(E_ITEM_POTION));
+					// Give new potion when the gamemode is not creative:
+					if (!a_Player.IsGameModeCreative())
+					{
+						auto & Inventory = a_Player.GetInventory();
+						auto NewItem = cItem(E_ITEM_POTION);
+						if (Inventory.ReplaceOneEquippedItem(NewItem) == 0)
+						{
+							// The potion didn't fit, toss it as a pickup:
+							a_Player.TossPickup(NewItem);
+						}
```

Comment:
This seems like it will be a pretty common construct. Add a function `cPlayer::ReplaceOneEquippedItemTossRest(const cItem &)` that does this (plus calculates the correct amount of leftover items to toss).